backend/internal/usecases/update_post_usecase.go

```diff
@@ -0,0 +1,41 @@
+package usecases
+
+import (
+	"errors"
+	"myapp/internal/controllers/schema"
+	"myapp/internal/entities"
+	"myapp/internal/exception"
+	"myapp/internal/interfaces"
+)
+
+type UpdatePostUsecase struct {
+	postRepository interfaces.PostRepository
+}
+
+func NewUpdatePostUsecase(
+	postRepository interfaces.PostRepository,
+) *UpdatePostUsecase {
+	return &UpdatePostUsecase{
+		postRepository: postRepository,
+	}
+}
+
+func (u *UpdatePostUsecase) Execute(request schema.PostRequest, userId int64, postId int64) (*entities.Post, error) {
+	post, err := u.postRepository.GetById(postId)
+	if err != nil {
+		return nil, err
+	} else if post == nil {
+		return nil, exception.ErrPostNotFound
+	}
+
+	// Check if the user is authorized to delete the post
+	if post.UserId != userId {
+		return nil, exception.ErrUnauthorizedToUpdatePost
+	}
+	updatedPost, err := u.postRepository.UpdatePost(request.Title, request.Body, userId)
+	if err == errors.New("post not found") {
+		return nil, exception.ErrPostNotFound
+	}
```

Comment:
ここのエラーハンドリングは以下で良さそう

```suggestion
	updatedPost, err := u.postRepository.UpdatePost(request.Title, request.Body, userId)
	if err != nil {
		return nil, err
	}
```

> 「repositoryのpost not foundをusecaseでErrorとして返したい」

この話やねんけど、どこでハンドリングするかって問題はあって悩みどころなのよね。
この usecase やと、repo.UpdatePost を呼び出す前に repo.GetById してるから、ビジネスロジック的には UpdatePost で "post not found" が返ってくることはありえへんねんな。
やから、ここで "post not found" が返ってきた場合は Internal ServerError が適切になるはずなんよ。

逆に、上の GetById はビジネスロジックとして、Post の存在を確認してるから "post not found" が発生したら専用のエラーを返したいって感じ